.github/workflows/ci.yml

```diff
@@ -38,6 +38,38 @@ jobs:
       - name: fast tests
         run: cargo test --workspace --locked
 
+      - name: Clean up the database
+        run: docker-compose down --volumes
+
+  build_test:
+    name: Test Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@master
+      - id: install
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: restore build & cargo cache
+        uses: Swatinem/rust-cache@v1
+
+      - name: Launch postgres and min.io
+        run: |
+          cp .env.sample .env
+          . .env
+          mkdir -p ${CRATESFYI_PREFIX}/public-html
+          docker-compose up -d db s3
+          # Give the database enough time to start up
+          sleep 5
+          # Make sure the database is actually working
+          psql "${CRATESFYI_DATABASE_URL}"
+
+      - name: Build
+        run: cargo build --workspace --locked
```

Comment:
I wonder if we could have a separate build step that outputs the binary somehow? That would avoid doubling the cache and CI time.

Reply:
in theory it's possible, I did a similar thing for a python project with 15 different jobs. 

What would be the binaries that `cargo test` needs? 
When it's the whole `/target` there is a chance that cache+load takes the same time as just restoring the rust cache + building it again.

Reply:
Hmm you could use `cargo build --bin --tests`, those shouldn't need any dependencies. I think cargo shows the paths of tests if you use --message-format json.

Reply:
related issue: https://github.com/rust-lang/cargo/issues/1924